src/components/CodeManagement/ManageCodesTab.jsx

```diff
@@ -0,0 +1,361 @@
+
+    if (couponId) {
+      couponRefs.forEach((coupon) => {
+        const { data: { id } } = coupon.props;
+        if (id !== parseInt(couponId, 10)) {
+          coupon.setCouponOpacity(true);
+        }
+      });
+    } else {
+      couponRefs.forEach((coupon) => {
+        coupon.setCouponOpacity(false);
+      });
+    }
+  }
+
+  removeQueryParams(keys) {
+    const queryParams = {};
+    keys.forEach((key) => {
+      queryParams[key] = undefined;
+    });
+    updateUrl(queryParams);
+  }
+
+  paginateCouponOrders(pageNumber) {
+    const page = pageNumber ? parseInt(pageNumber, 10) : 1;
+    this.props.fetchCouponOrders({ page });
+  }
+
+  handleRefreshData() {
+    this.paginateCouponOrders(1);
+    this.removeQueryParams(['coupon_id', 'page', 'overview_page']);
+    this.setState({ searchQuery: '' });
+  }
+
+  handleCouponExpand(selectedIndex) {
+    const coupons = this.getCouponRefs();
+    const selectedCoupon = coupons[selectedIndex];
+    const couponId = selectedCoupon.props.data.id;
+    const queryParams = {
+      coupon_id: couponId,
+    };
+    updateUrl(queryParams);
+    this.setCouponOpacity(couponId);
+    this.setState({ searchQuery: '' });
+  }
+
+  handleCouponCollapse() {
+    this.setCouponOpacity();
+    this.removeQueryParams(['coupon_id', 'page']);
+  }
+
+  hasCouponData(coupons) {
+    if (!coupons) {
+      return false;
+    }
+    const { results } = coupons;
+    return results && results.length > 0;
+  }
+
+  renderLoadingMessage() {
+    return <LoadingMessage className="coupons mt-3" />;
+  }
+
+  renderErrorMessage() {
+    return (
+      <Alert variant="danger" icon={Info}>
+        <Alert.Heading>Unable to load coupons</Alert.Heading>
+        <p>Try refreshing your screen ({this.props.error.message})</p>
+      </Alert>
+    );
+  }
+
+  renderCoupons() {
+    const { coupons, location } = this.props;
+    const queryParams = new URLSearchParams(location.search);
+
+    return (
+      <>
+        {coupons.results.map((coupon, index) => (
+          <Coupon
+            ref={(node) => { this.couponRefs[index] = node; }}
+            key={coupon.id}
+            data={coupon}
+            isExpanded={coupon.id === parseInt(queryParams.get('coupon_id'), 10)}
+            onExpand={() => this.handleCouponExpand(index)}
+            onCollapse={() => this.handleCouponCollapse()}
+          />
+        ))}
+        <div className="d-flex mt-4 justify-content-center">
+          <Pagination
+            onPageSelect={page => updateUrl({
+              coupon_id: undefined,
+              page: undefined,
+              overview_page: page !== 1 ? page : undefined,
+            })}
+            pageCount={coupons.num_pages}
+            currentPage={coupons.current_page}
+            paginationLabel="coupons pagination"
+          />
+        </div>
+      </>
+    );
+  }
+
+  renderRequestCodesSuccessMessage() {
+    return (
+      <Alert variant="success" icon={CheckCircle} dismissible>
```

Comment:
The `StatusAlert` components were swapped for `Alert`, as `StatusAlert` component is deprecated in Paragon.